### PR TITLE
Update file upload documentation

### PR DESCRIPTION
**Pull request by marcw**
There was a problem with content-type. A deserializer is needed.

### CODE REVIEW

**Comment by Simperfit** [comment]
I'm going to test it today, thanks @marcw!

File: core/file-upload.md [hunk 1]
```diff
@@ -171,23 +173,111 @@ final class MediaObjectType extends AbstractType
...
     }
+
+    public function getBlockPrefix()
+    {
+        return '';
+    }
+}
+```
+
+## Deserializing the `multipart/form-data` content type.
```
[the remark this answers is not in the document]
Comment (marcw) [reply core/file-upload.md]: @teohhanhui It doesn't help. What do you suggest?

File: core/file-upload.md [hunk 2]
```diff
@@ -48,16 +48,17 @@ use ApiPlatform\Core\Annotation\ApiResource;
...
  * @ORM\Entity
  * @ApiResource(iri="http://schema.org/MediaObject", collectionOperations={
  *     "get",
  *     "post"={
- *         method="POST",
- *         path="/media-objects",
- *         controller=CreateMediaObjectAction::class,
+ *         "method"="POST",
+ *         "path"="/media_objects",
+ *         "controller"=CreateMediaObjectAction::class,
```
Comment (teohhanhui) [inline core/file-upload.md]: Add this:

```php
 *         "controller"=CreateMediaObjectAction::class,
 *         "defaults"={"_api_receive"=false},
 *     },
 */
```

File: core/file-upload.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (teohhanhui) [reply core/file-upload.md]: See https://github.com/api-platform/docs/pull/494#discussion_r193066512 :smile:

File: core/file-upload.md (hunk 2 above)
Comment (teohhanhui) [reply core/file-upload.md]: Basically, it disables the `ReadListener`, `DeserializeListener` and `ValidateListener`.

**Comment by marcw** [comment]
updated! :)

File: core/file-upload.md [hunk 3]
```diff
@@ -171,23 +177,29 @@ final class MediaObjectType extends AbstractType
...
+
+    public function getBlockPrefix()
+    {
+        return '';
+    }
 }
 ```
 
-## Making a Request to The `/media-objects` Endpoint
+## Making a request to the `/media_objects` endpoint.
```
Comment (dunglas) [inline core/file-upload.md]: Should be reverted (except the uppercase `T`), we use title case for titles.

File: core/file-upload.md [hunk 4]
```diff
@@ -171,23 +177,29 @@ final class MediaObjectType extends AbstractType
...
+
+    public function getBlockPrefix()
+    {
+        return '';
+    }
 }
 ```
 
-## Making a Request to The `/media-objects` Endpoint
+## Making a Request to the `/media_objects` Endpoint.
```
Comment (teohhanhui) [inline core/file-upload.md]: The period (`.`) at the end should be removed.

**Pull request merged**